src/lib/MedStickerEncryptionCharlie.js

```diff
@@ -0,0 +1,80 @@
+import create from "./factory";
+import { arrayBufferToHex } from "./utilities";
+import { CHARLIE } from "./constants";
+
+const scrypt = create("scrypt");
+const gcm = create("AES-GCM");
+
+/**
+ * @param secret {string}
+ * @param salt {string}
+ * @returns {Promise<CryptoKeyPair | CryptoKey>}
+ */
+export function hash(secret, salt) {
+  try {
+    return scrypt.generateKey(secret, salt);
+  } catch (e) {
+    throw new Error("HashFailed");
+  }
+}
+
+/**
+ * Given a hashed string, returns a concatenated string
+ * that contains the version of the encryption its being
+ * used and the incoming string in hex format
+ * @param hashed {string}
+ * @returns {string}
+ */
+
+export function fingerprintSecret(hashed) {
```

Comment:
its not only secret fingerprint

Reply:
afaic this has been addressed. please resolve or correct me if i am worng